src/linker/Linker/ModuleDefinitionExtensions.cs

```diff
@@ -10,5 +10,25 @@ public static bool IsCrossgened (this ModuleDefinition module)
 			return (module.Attributes & ModuleAttributes.ILOnly) == 0 &&
 				(module.Attributes & ModuleAttributes.ILLibrary) != 0;
 		}
+
+		public static TypeDefinition ResolveType (this ModuleDefinition module, string typeFullName)
```

Comment:
Nit: Add a comment or maybe even an assert that the `typeFullName` must be a simple type name, no generics, arrays and so on.

Reply:
Shouldn't `Module.GetType` work fine with either of those?

Reply:
`Module.GetType` will only works for type definitions - so for open generics. Generic instantiations work work. Similarly for arrays, pointers and so on. That's OK, the places where we call the new method handle those cases and only calls this for "type defs", I just wanted to make sure that we don't make that mistake in the future.